types/react/index.d.ts

```diff
@@ -92,7 +93,7 @@ declare namespace React {
         ref?: Ref<T>;
     }
     interface ClassAttributes<T> extends Attributes {
-        ref?: LegacyRef<T>;
+        ref?: LegacyRef<T | undefined>;
```

Comment:
This makes writing callback refs more difficult because you now have to check for an unreachable value.

Reply:
You have to write 

```tsx
const ref = useRef<HTMLDivElement>()
let myVal:HTMLDivElement|undefined|null
const div = <div ref={v => myVal = v}/>
```

That's not very hard, and above all, it's valid, since myVal can be null/undefined. If what you're complaining about is this:

```tsx
const ref = useRef<HTMLDivElement>()
let myVal:HTMLDivElement|null = null
const div = <div ref={v => myVal = v}/>
```

...now getting an error. Then yes, that's a flaw. If that's your beef that seems totally solvable by a better structure around the different ref types (LegacyRef, RefObject, function etc.)

Reply:
Could be fixed e.g. with 
```ts
type LegacyRef<T> = string | Ref<T> | {current:undefined}
```